spark3/src/main/java/org/apache/iceberg/spark/source/StreamingOffset.java

```diff
@@ -132,4 +130,18 @@ public String toString() {
     return String.format("Streaming Offset[%d: position (%d) scan_all_files (%b)]",
                 snapshotId, position, scanAllFiles);
   }
+
+  private static StreamingOffset fromJsonNode(JsonNode node) {
+    // The version of StreamingOffset. The offset was created with a version number
+    // used to validate when deserializing from json string.
+    int version = JsonUtil.getInt(VERSION, node);
+    Preconditions.checkArgument(version == CURR_VERSION,
+        "Cannot parse offset JSON: offset version %s is not supported", version);
```

Comment:
Should add "This version of Iceberg only supports version CURR_VERSION"

Reply:
added this.